src/lib/url/url.hpp

```diff
@@ -92,6 +93,116 @@ class Url : public otUrl
      *
      */
     const char *GetProtocol(void) const { return mProtocol; }
+
+    /**
+     * Indicates whether or not the url contains the parameter.
+     *
+     * @param[in]  aName  A pointer to the parameter name.
+     *
+     * @retval TRUE   The url contains the parameter.
+     * @retval FALSE  The url doesn't support the parameter.
+     *
+     */
+    bool DoesParamExist(const char *aName) const { return (GetValue(aName) != nullptr); }
+
+    /**
+     * Gets a `uint32_t` parameter value.
+     *
+     * The parameter value in string is parsed as decimal or hex format (if contains `0x` or `0X` prefix).
+     *
+     * @param[in]  aName    A pointer to the parameter name.
+     * @param[out] aValue   A reference to an `uint32_t` variable to output the parameter value.
+     *                      The original value of @p aValue won't change if failed to get the value.
+     *
+     * @retval OT_ERROR_NONE          The parameter value was parsed successfully.
+     * @retval OT_ERROR_NOT_FOUND     The parameter name was not found.
+     * @retval OT_ERROR_INVALID_ARGS  The parameter value was not contain valid number (e.g., value out of range).
+     *
+     */
+    otError GetUint32(const char *aName, uint32_t &aValue) const;
```

Comment:
Suggestion here: It is ok as it is, but how about renaming `ParseUint32()` here and similarly for all methods below?

I suggest this, because we generally use this naming convention of `Get{SomeProp}()` for simpler getters of properties. This method is doing more work than a typical getter, as it searches the URL for `aName` and then parses its associated value.

Reply:
Thanks.